pkg/daemon/daemon.go

```diff
@@ -119,19 +127,158 @@ func (d *daemon) Run() {
 }
 
+// If network identified by networkID is IbSriov return network name and spec
+func (d *daemon) GetIbSriovNetwork(networkID string) (string, *utils.IbSriovCniSpec, error) {
+	log.Info().Msgf("processing network networkID %s", networkID)
+	networkNamespace, networkName, err := utils.ParseNetworkID(networkID)
+	if err != nil {
+		return "", nil, fmt.Errorf("failed to parse network id %s with error: %v", networkID, err)
+	}
+	netAttInfo, err := d.kubeClient.GetNetworkAttachmentDefinition(networkNamespace, networkName)
+	if err != nil {
+		return "", nil, fmt.Errorf("failed to get networkName attachment %s with error: %v", networkName, err)
+	}
+
+	log.Debug().Msgf("networkName attachment %v", netAttInfo)
+	networkSpec := make(map[string]interface{})
+	err = json.Unmarshal([]byte(netAttInfo.Spec.Config), &networkSpec)
+	if err != nil {
+		return "", nil, fmt.Errorf("failed to parse networkName attachment %s with error: %v", networkName, err)
+	}
+	log.Debug().Msgf("networkName attachment spec %+v", networkSpec)
+
+	ibCniSpec, err := utils.GetIbSriovCniFromNetwork(networkSpec)
+	if err != nil {
+		return "", nil, fmt.Errorf("failed to get InfiniBand SR-IOV CNI spec from network attachment %+v, with error %v",
+			networkSpec, err)
+	}
+
+	log.Debug().Msgf("CNI spec %+v", ibCniSpec)
+	return networkName, ibCniSpec, nil
+}
+
+// Get Pod's IB network specified by name. Update netMap with Pod's networks.
+func getIbSriovPodNetwork(netName string, pw *PodWrapper, netMap map[types.UID][]*v1.NetworkSelectionElement) error {
+	var err error
+	networks, ok := netMap[pw.pod.UID]
+	if !ok {
+		networks, err = netAttUtils.ParsePodNetworkAnnotation(pw.pod)
+		if err != nil {
+			return fmt.Errorf("failed to read pod networkName annotations pod namespace %s name %s, with error: %v",
+				pw.pod.Namespace, pw.pod.Name, err)
+		}
+
+		netMap[pw.pod.UID] = networks
+	}
+	network, err := utils.GetPodNetwork(networks, netName)
+	if err != nil {
+		return fmt.Errorf("failed to get pod networkName spec %s with error: %v", netName, err)
+	}
+
+	pw.ibNetwork = network
+	pw.networks = networks
+	return nil
+}
+
+// Verify if GUID already exist for given network ID and allocates new one if not
+func (d *daemon) mapGUIDToPodNetwork(allocatedGUID, podNetworkID string, podUID types.UID) error {
+	if _, exist := d.guidPodNetworkMap[allocatedGUID]; exist {
+		if podNetworkID != d.guidPodNetworkMap[allocatedGUID] {
+			return fmt.Errorf("failed to allocate requested guid %s, already allocated for %s",
+				allocatedGUID, d.guidPodNetworkMap[allocatedGUID])
+		}
+	} else if err := d.guidPool.AllocateGUID(allocatedGUID); err != nil {
+		return fmt.Errorf("failed to allocate GUID for pod ID %s, wit error: %v", podUID, err)
+	} else {
+		d.guidPodNetworkMap[allocatedGUID] = podNetworkID
+	}
+
+	return nil
+}
+
+// Allocate GUID if network don't have it and update Pod's networks annotation
+func (d *daemon) ProceedNetworkGUID(pw *PodWrapper, networkID string) error {
```

Comment:
you mean "ProceedNetworkGUID" ?

Reply:
Yes?

Reply:
Sorry I mean Do you mean "ProcessNetworkGUID"?  the method name is not clear to me

Reply:
Yes, "Process" sounds better.